### PR TITLE
Manager is not element

### PR DESCRIPTION
**Pull request by clero**
Avoid ParameterMgr inheritance to Element
Allows to avoid tree structure guess work

### CODE REVIEW

**Comment by clero** [comment]
@dawagner @krocard please review

File: parameter/ParameterMgr.cpp [hunk 1]
```diff
@@ -657,16 +650,19 @@ bool CParameterMgr::loadSettingsFromConfigFile(string& strError)
...
+                strXmlConfigurationDomainsFilePath, strXmlConfigurationDomainsFolder,
+                EParameterConfigurationLibrary, "SystemClassName")) {
 
         return false;
     }
     // We have loaded the whole system structure, compute checksum
-    const CSystemClass* pSystemClass = getConstSystemClass();
-    _uiStructureChecksum = pSystemClass->computeStructureChecksum() + getConfigurableDomains()->computeStructureChecksum() + getSelectionCriteria()->computeStructureChecksum();
+    _uiStructureChecksum = mSystemClass.computeStructureChecksum() +
+                           mDomains.computeStructureChecksum();
```
Comment (dawagner) [inline parameter/ParameterMgr.cpp]: CSelectionCriterionLibrary::getKind() is now dead code.

**Comment by dawagner** [comment]
the decrease in coverage is only related to dead getKind() methods that can't be removed yet.

File: parameter/ParameterType.cpp [hunk 2]
```diff
@@ -98,7 +99,7 @@ void CParameterType::showProperties(string& strResult) const
     }
 
     // Scalar size
-    strResult += "Scalar size: " + toString(getSize()) + " byte(s) \n";
+    strResult += "Scalar size: " +CUtility::toString(getSize()) + " byte(s) \n";
```
Comment (dawagner) [inline parameter/ParameterType.cpp]: coding style: missing whitespace before "+"

**Comment by dawagner** [comment]
Please resubmit against the master branch :) ok for me after once you fix the minor coding style issue in parameter/ParameterType.cpp.

Thanks a lot, @clero.

@OznOg Please review.

File: parameter/ParameterType.cpp (hunk 2 above)
Comment (clero) [inline parameter/ParameterType.cpp]: done

**Comment by clero** [comment]
closed for master submission

**Comment by clero** [comment]
@dawagner  in fact maybe it can be good to send it on next branch as it already has been updated to the new log API
Let me know if you still prefer a submission on master branch.

**Comment by clero** [comment]
getKind must be removed from ParameterMgr!

**Pull request merged**